packages/manager/src/features/Images/EditImageDrawer.tsx

```diff
@@ -0,0 +1,161 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { APIError, Image, UpdateImagePayload } from '@linode/api-v4';
+import { updateImageSchema } from '@linode/validation';
+import * as React from 'react';
+import { Controller, useForm } from 'react-hook-form';
+
+import { ActionsPanel } from 'src/components/ActionsPanel/ActionsPanel';
+import { Drawer } from 'src/components/Drawer';
+import { Notice } from 'src/components/Notice/Notice';
+import { TagsInput } from 'src/components/TagsInput/TagsInput';
+import { TextField } from 'src/components/TextField';
+import { useUpdateImageMutation } from 'src/queries/images';
+
+import { useImageAndLinodeGrantCheck } from './utils';
+
+interface Props {
+  image?: Image;
+  onClose: () => void;
+  open: boolean;
+}
+export const EditImageDrawer = (props: Props) => {
+  const { image, onClose, open } = props;
+
+  const { canCreateImage } = useImageAndLinodeGrantCheck();
+
+  const defaultValues = {
+    description: image?.description ?? undefined,
+    label: image?.label,
+    tags: image?.tags,
+  };
+
+  const {
+    control,
+    formState,
+    handleSubmit,
+    reset,
+    setError,
+  } = useForm<UpdateImagePayload>({
+    defaultValues,
+    mode: 'onBlur',
+    resolver: yupResolver(updateImageSchema),
+    values: defaultValues,
+  });
+
+  const { mutateAsync: updateImage } = useUpdateImageMutation();
+
+  React.useEffect(() => {
+    if (open) {
+      reset();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [open]);
+
+  const onSubmit = handleSubmit(async (values) => {
+    if (!image) {
+      return;
+    }
+
+    await updateImage({ imageId: image.id, ...values })
+      .then(onClose)
+      .catch((errors: APIError[]) => {
+        for (const error of errors) {
+          if (
+            error.field === 'label' ||
+            error.field == 'description' ||
+            error.field == 'tags'
+          ) {
+            setError(error.field, { message: error.reason });
+          } else {
+            setError('root', { message: error.reason });
+          }
+        }
+      });
+  });
+
+  return (
+    <Drawer onClose={onClose} open={open} title="Edit Image">
+      {!canCreateImage && (
+        <Notice
+          text="You don't have permissions to edit images. Please contact an account administrator for details."
+          variant="error"
+        />
+      )}
+
+      {formState.errors.root?.message && (
+        <Notice
+          spacingBottom={8}
+          text={formState.errors.root.message}
+          variant="error"
+        />
+      )}
+
+      <Controller
+        render={({ field, fieldState }) => (
+          <TextField
+            data-qa-image-label
+            disabled={!canCreateImage}
+            error={Boolean(fieldState.error)}
+            errorText={fieldState.error?.message}
+            label="Label"
+            onChange={(e) => field.onChange(e.target.value)}
+            value={field.value}
```

Comment:
We should make sure we pass an `onBlur` to all text fields to that validation happens at the right times
```suggestion
            value={field.value}
            onBlur={field.onBlur}
```

Without it, no validation happens when I leave the field before the first submit

https://github.com/linode/manager/assets/115251059/fba70a3b-7ef1-4fd1-a502-036066d248c8